### PR TITLE
Python: reorganize expressions module structure

### PR DESCRIPTION
**Pull request by dramaticlly**
This help with https://github.com/apache/iceberg/issues/4435 to provide better structure under the expressions module.

Rename old expression module to include both literals and operations

This is mechanical refactor without logical change. I also want to extract the base class out of literals but looks like there's another PR #4466  already doing so, so to minimize the conflict I will leave this PR as reorganization only

CC @samredai  @rdblue

### CODE REVIEW

File: python/tests/expressions/test_operations.py [hunk 1]
```diff
@@ -17,26 +17,26 @@
 
 import pytest
 
-from iceberg import expressions
+from iceberg.expressions import operations
```
Comment (rdblue) [inline python/tests/expressions/test_operations.py:20]: Shouldn't `operations` go into `base.py` and then get exposed in `__init__.py`? I would probably leave the tests unchanged and use `expressions.Operation.*`.

Comment (samredai) [reply python/tests/expressions/test_operations.py:20]: +1 that operations should go into base.py but would it be helpful for the tests to use the full imports? In other words `from iceberg.expressions import base`. It makes it a bit easier to jump from looking at a test to the relevant source code.

**Comment by dramaticlly** [comment]
moved `operations` into `base.py` and leave the unit test coverage for operations unchanged.

@rdblue  and @samredai  appreciate another look

File: python/tests/expressions/test_operations.py (hunk 1 above)
Comment (dramaticlly) [reply python/tests/expressions/test_operations.py]: Thank you, updated per your suggestions!

**Review by samredai — APPROVED** [review]
LGTM. One small nit though is we should now rename `test_operations.py` to `test_expressions_base.py`. The pattern we've been using so far is to keep the tests directory in sync with the src directory, but `expressions/test_base.py` will run into a conflict with `io/test_base.py` (pytest wants unique filenames). We should probably also rename `io/test_base.py` to `io/test_io_base.py`.

Thanks @dramaticlly!

**Comment by dramaticlly** [comment]
Thank you @samredai  for your quick review, I renamed
- test_expressions_base 
- test_io_base.py

as suggested!

**Review by rdblue — APPROVED** [review]

**Comment by rdblue** [comment]
Thanks, @dramaticlly!

**Pull request merged**